pkg/security/probe/probe_linux.go

```diff
@@ -979,26 +979,53 @@ func (p *Probe) SetApprovers(eventType eval.EventType, approvers rules.Approvers
 	}
 
 	for _, newApprover := range newApprovers {
-		seclog.Tracef("Applying approver %+v", newApprover)
+		seclog.Tracef("Applying approver %+v for event type %s", newApprover, eventType)
 		if err := newApprover.Apply(p.Manager); err != nil {
 			return err
 		}
+
+		tags := []string{
+			fmt.Sprintf("approver_type:%s", getApproverType(newApprover.GetTableName())),
+			fmt.Sprintf("event_type:%s", eventType),
+		}
+
+		if err := p.StatsdClient.Incr(metrics.MetricApproverAdded, tags, 1.0); err != nil {
+			seclog.Tracef("couldn't increment MetricApproverAdded metric: %s", err)
+		}
 	}
 
 	if previousApprovers, exist := p.approvers[eventType]; exist {
 		previousApprovers.Sub(newApprovers)
 		for _, previousApprover := range previousApprovers {
-			seclog.Tracef("Removing previous approver %+v", previousApprover)
+			seclog.Tracef("Removing previous approver %+v for event type %s", previousApprover, eventType)
 			if err := previousApprover.Remove(p.Manager); err != nil {
 				return err
 			}
+
+			tags := []string{
+				fmt.Sprintf("approver_type:%s", getApproverType(previousApprover.GetTableName())),
+				fmt.Sprintf("event_type:%s", eventType),
+			}
+			if err := p.StatsdClient.Decr(metrics.MetricApproverAdded, tags, 1.0); err != nil {
```

Comment:
I would rather declare a counter line 980 that I would increment line ~986 and decrement line ~100 and I would call the statsdclient line ~1014 with this counter

Reply:
hmmm we'd have to have a counter for every combination of tags